### DISCUSSION TITLE
ECU-B Connection shutdown after few days

### DISCUSSION THREAD

**Discussion by Scorp1350** [ksheumaker/homeassistant-apsystems_ecur#182] [Q&A] [answered]
Hi, 

First of all thank for this amazing job !

I'm using it with and ECU-B, I'm having trouvle with "I think" connection, after few days of well working the data collect stop and I must restart HA to collect data again.

Any speciale config or trick ?

thank you

**Comment by HAEdwin** [comment]
A restart of HA is not necessary. It is sufficient to give the ECU a powercycle and turn the switch.ecu_query_device back on.
It is therefor useful to place the switch on the dashboard and use it for notification on, for example, the telephone if it ends up being switched off. The switch can also be used to automatically powercycle the ECU using a smartplug.

Query the ECU no more than every five minutes.

It is unclear where APsystems wants to go with the firmware on the ECU-B. The way of installing using the EMA app on the phone is probably not going to be easy because it uses the same principle. Depending on the firmware versions the stability results might differ.

We've done a lot of tweaking to try and make it as stable as possible for all ECU models but the fact is that the method used is not officially the way the ECU should be used, only newer ECU-R's and ECU-C's have a modbus to retreive data from the ECU.

Maybe ECU-B owners can team up and experiment - exchange experiences - tweaking the code.

**Reply by Scorp1350** [reply]
Thank you for the reply and explanation, It's disapointing that APSys give it as a black box.

By the way, when I restart the HA I don't need to powercycle the ECU, maybe just playing with the switch could be enough ?! I'll try

**Comment by Scorp1350** [comment, marked as answer]
Hi @HAEdwin 

Following you're advise I ste a routine to toogle the switch before the sunrise, for about 4-5 days I've nore more connection issue, It's stable and no need to restart the ECU-B

Thank you

**Reply by HAEdwin** [reply]
The switch was introduced when we noticed that the ECU did not respond well on queries whilst being busy with maintenance. So at that time querying the ECU was avoided. In later firmware versions we noticed an improvement so we could query 24/7 again (like I still do these days). But it seems to have returned. I'll close this issue, might any other question arrise, feel free to open a new ticket.
